R/covr.R

```diff
@@ -588,21 +608,16 @@ merge_coverage.list <- function(coverage_objs) {
 
   x <- coverage_objs[[1]]
   names <- names(x)
+  clean_coverage_tests(x)
 
   for (y in tail(coverage_objs, -1L)) {
-    # align tests from coverage objects
-    test_idx <- match(names(y$tests), Filter(nchar, names(x$tests)))
-    new_test_idx <- if (!length(test_idx)) seq_along(y$tests) else which(is.na(test_idx))
-    test_idx[new_test_idx] <- length(x$tests) + seq_along(new_test_idx)
 
-    # append any tests that we haven't encountered in previous objects
-    x$tests <- append(x$tests, y$tests[new_test_idx])
-    y$tests <- NULL
+    # only affects coverage produced with options(covr.record_tests = TRUE)
+    clean_coverage_tests(y)
+    merge_coverage_tests(from = y, into = x)  # modifies environments 'x' and 'y' in-place
 
```

Comment:
In the spirit of keeping the "vanilla" `covr` workflow clean, I tried to refactor out a lot of the test recording behaviors into separate functions. This is a bit of a theme with some of the changes in this PR.